### PR TITLE
[WIP] Increase test Coverage pkg aaparser

### PR DESCRIPTION
**Pull request by raja-sami-10p**
Signed-off-by: Raja Sami <raja.sami@tenpearls.com>

**- What I did**
Increase test Coverage pkg aaparser

**- How to verify it**
```sh
make test-unit
```

**- A picture of a cute animal (not mandatory but encouraged)**
:shark: :

### CODE REVIEW

File: pkg/aaparser/aaparser_test.go [hunk 1]
```diff
@@ -71,3 +86,33 @@ Copyright 2009-2012 Canonical Ltd.
...
+
+func requiresRoot(t *testing.T) {
+	if !rootEnabled {
+		t.Skip("skipping test that requires root")
+		return
+	}
+	require.Equal(t, 0, os.Getuid(), "This test must be run as root.")
+}
+
+func TestGetVersion(t *testing.T) {
```
Comment (aaronlehmann) [inline pkg/aaparser/aaparser_test.go]: This is not actually testing the code. A better test would call `parseVersion` with various test cases and check the reutrn values.

File: pkg/aaparser/aaparser_test.go [hunk 2]
```diff
@@ -71,3 +86,33 @@ Copyright 2009-2012 Canonical Ltd.
...
+	output, err := cmd("", "--version")
+	require.NotNil(t, output)
+	version, err := GetVersion()
+	require.NoError(t, err)
+	require.NotEqual(t, -1, version)
+}
+
+func TestLoadProfile(t *testing.T) {
```
Comment (aaronlehmann) [inline pkg/aaparser/aaparser_test.go]: This unit test invokes the external program `apparmor_parser`, which I believe has side effects on the system's apparmor profiles. I don't think a unit test should depend on external binaries, and it especially shouldn't reconfigure the system where it's run.

Finally, this test isn't actually checking that the profile was loaded correctly.

I think apparmor should be tested as part of integration tests, and I believe there is already coverage for this.

Comment (raja-sami-10p) [inline pkg/aaparser/aaparser_test.go]: Agreed, and given that we already have integration test for this in place, I will close this PR.

File: pkg/aaparser/aaparser_test.go (hunk 1 above)
Comment (raja-sami-10p) [inline pkg/aaparser/aaparser_test.go]: There is already a test doing just what you have mentioned.

**Pull request closed without merging**